packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -335,14 +344,14 @@ class LinearGradient extends Gradient {
 
   @override
   Gradient lerpFrom(Gradient a, double t) {
-    if (a == null || (a is LinearGradient && a.colors.length == colors.length)) // TODO(ianh): remove limitation
```

Comment:
This todo also exists on the other Gradient subclasses in this file. I believe they can be removed there as well, no?